core/src/main/java/feign/template/Template.java

```diff
@@ -214,7 +216,7 @@ private void parseFragment(String fragment) {
       /* check to see if we have an expression or a literal */
       String chunk = tokenizer.next();
 
-      if (chunk.startsWith("{")) {
+      if (chunk.startsWith("{") && !isValidJsonLiteral(fragment)) {
```

Comment:
@velo This is the fix I think we need to resolve this issue. Check if the string is a json literal before parsing it as a template. So if you agree with this solution and I have to add this check here, what would be the best approach please? Open for suggestions.

Reply:
What about the regex option you proposed?

Reply:
I can try. The regex looks a bit complicated and yet to confirm if it breaks the template string.  Let me add that with a few tests and update the PR.

Reply:
feel free to pick any other name besides `expression`

Reply:
I prefer this approach. Let’s add a property to the header annotation indicating that it should be treated like a literal. What about naming the property “literal” and be a Boolean?

Reply:
It will avoid any regular expressions and simplify the logic to be if “literal” is true, just render otherwise, resolve the expression. There should be a “Literal” type I think in the template package. I think.

Reply:
Please remember annotation isn't the only way of setting the header

Reply:
I agree. This approach is better and avoid any assumption about the expression/literal.

Reply:
Although adding a variable like literal in Headers annotation wouldn't always be right since headers values is a string array, it's possible one of header has an expression and one is a literal. A variable would only work out if we can tailor the new variable for every header added to the array. That will be an expensive change.